test/unit/StakingProducts/recalculateEffectiveWeight.js

```diff
@@ -0,0 +1,447 @@
+const { ethers } = require('hardhat');
+const { expect } = require('chai');
+
+const { parseEther } = ethers.utils;
+const { Zero, One } = ethers.constants;
+
+const {
+  allocateCapacity,
+  depositTo,
+  burnStake,
+  setStakedProducts,
+  daysToSeconds,
+  burnStakeParams,
+  buyCoverParamsTemplate,
+  newProductTemplate,
+} = require('./helpers');
+const { setEtherBalance } = require('../../utils/evm');
+const { increaseTime } = require('../../utils').evm;
+
+const DEFAULT_PRODUCTS = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20];
+describe('recalculateEffectiveWeight', function () {
+  it('should return 0 for products not found in stakingPool', async function () {
```

Comment:
What is being tested here is the function `recalculateEffectiveWeights`

`Should skip products not found in staking pool`

Or something to that effect because this unit test is about a different function aka recalculateEffectiveWeights that does not return anything.

getProduct returns, getTotalEffectiveWeight returns etc.